lib/opentelemetry/trace/exporter/http_client.lua

```diff
@@ -40,14 +40,16 @@ function _M.do_request(self, body)
     if not res then
         ngx.log(ngx.ERR, "request failed: ", err)
         httpc:close()
-        return
+        return nil, err
     end
 
     if res.status ~= 200  then
```

Comment:
We may want to get more specific with our handling of different HTTP status codes.